### PR TITLE
Rename ./specification to ./docs

### PR DESCRIPTION
**Pull request by chalin**
Hi @tigrannajaryan @svrnm @cartermp et al.,

Why this PR? Because the documented OTLP specification is strictly contained within `./specification/otlp.md`. The remaining pages under `./specification` are supporting documentation. So, in this PR I propose the following:

- Rename `./specification` to `./docs`
- Rename `specification/otlp.md` to `./docs/specification.md`
- Adjusts links in other .md pages

Thoughts?

Contributes to:
 
- https://github.com/open-telemetry/opentelemetry.io/issues/2642

/cc @open-telemetry/specs-approvers

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
Is this necessary for website publishing or you are just proposing an improvement in the structure?

**Comment by chalin** [comment]
Both (since they are related).

File: README.md
Comment (tigrannajaryan) [inline README.md:59]: Nit: unrelated change.

**Review by tigrannajaryan — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

File: README.md
Comment (chalin) [reply README.md:59]: My editor strips off trailing whitespace on file save. Given that this is a nit, I'm assuming it's ok for it to be merged too :)

Comment (chalin) [reply README.md:59]: If so, can you mark this as resolved @tigrannajaryan? Thanks.

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers please review, this blocks website work.

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
lgtm

**Comment by chalin** [comment]
On hold until the following is resolved:

- #471

**Comment by chalin** [comment]
- Off hold, since #471 has been merged
- Rebased
- All checks are passing
- Note that I also committed Hugo front matter (via HTML comments) at the top of the doc pages (similar to what is done for the OTel spec)

**Comment by tigrannajaryan** [comment]
@chalin please resolve the conflicts.

**Comment by chalin** [comment]
> @chalin please resolve the conflicts.

@tigrannajaryan - done:
image

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
@chalin Sorry for the late review ;(

**Pull request merged**
